sleap/gui/app.py

```diff
@@ -1120,6 +1145,35 @@ def _update_gui_state(self):
 
         # Update menus
 
+        def _update_sessions_menu(self, frame_idx):
+            """Update the instance groups menu based on the selected frame index in the session."""
+            session = self.state.get("session")
+            if session is None:
+                return 
+
+
+            frame_idx = self.state["frame_idx"]
+            frame_group = session.frame_groups.get(frame_idx, None)
+            if frame_group is not None:
+                for inst_group_ind, instance_group in enumerate(frame_group.instance_groups):
+                    if inst_group_ind < 9:
+                        key_command = Qt.SHIFT + Qt.Key_0 + inst_group_ind + 1
+                    else:
+                        key_command = ""
+        
+                    self.inst_groups_menu.addAction(
+                        instance_group.name,
+                        lambda x=instance_group: self.commands.setInstanceGroup(x),
+                        shortcut=key_command
+                    )
+    
+            self.inst_groups_menu.addAction(
+                "New Instance Group",
+                self.commands.addInstanceGroup,
+                shortcut=Qt.SHIFT + Qt.Key_0
+            )
```

Comment:
Move this entire function outside of the `_update_gui_state` method (preferably placed right below `MainWindow._update_tracks_menu`). We want `_update_sessions_menu` to be a method of the `MainWindow` class, but currently, it is hidden inside `MainWindow._update_gui_state`.